### PR TITLE
make phoebus a library

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

Similar to what we do in riot. This has several advantages:
1. Unit test code can link against Phoebus, significantly simplifying the cmake workflow
2. Other codes can slurp in Phoebus, if this is desired.

@brryan and @bprather this might be useful for you. And @mari2895 this should solve the issue in #205 .

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by calling `scripts/bash/format.sh`.
- [ ] Explain what you did.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -150,13 +152,15 @@ add_executable(phoebus
 
   tracers/tracers.hpp
   tracers/tracers.cpp
+
+  # Generated
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/generated>/compile_constants.hpp
 )
```
Comment (mari2895) [inline src/CMakeLists.txt:158]: Thanks for adding this! you might want to do the same thing for `test_utils.hpp` maybe cause tests are still failing.

Comment (Yurlungur) [reply src/CMakeLists.txt:158]: `test_utils` doesn't belong in `phoebuslib` but I re-added it to the relevant unit tests.

**Review by mari2895 — APPROVED** [review]
LGTM, thanks!

**Review by lroberts36 — APPROVED** [review]
LGTM

**Review by AstroBarker — APPROVED** [review]
All tests now passing locally, LGTM

**Pull request merged**